product_docs/docs/pgd/5/admin-tpa/upgrading_major_rolling.mdx

```diff
@@ -0,0 +1,510 @@
+
+!!! Note
+Some steps of this process will involve you running commands as the Postgres owner. We will refer to this user as `postgres` throughout when appropriate. If you are running EDB Postgres Advanced Server, you will need to substitute that user with `enterprisedb` in all relevant commands.
+!!!
+
+### Confirm the current Postgres version
+
+Connect to any node that is still running Postgres 15 and use the following command to connect to PGD's `bdrdb` database:
+
+```bash
+sudo -u postgres psql bdrdb
+```
+Then, when connected to bdrdb, use the following query:
+
+```sql
+SELECT * FROM bdr.group_versions_details;
+```
+
+You should see output similar to this for your cluster:
+
+```
+  node_id   | node_name |        postgres_version        | bdr_version 
+------------+-----------+--------------------------------+-------------
+ 3490219809 | kaftan    | 15.5 (Debian 15.5-1EDB.buster) | 5.3.0
+ 2710197610 | kaboom    | 15.5 (Debian 15.5-1EDB.buster) | 5.3.0
+ 2111777360 | kaolin    | 15.5 (Debian 15.5-1EDB.buster) | 5.3.0
+(3 rows)
+```
+
+Confirm that the Postgres version is the expected version. Then, use the `exit` command to return to the node's command prompt.
+
+### Plan upgrade order
+
+The cluster must be available throughout the process (i.e., a *rolling* upgrade). To achieve continuous cluster availability, we must perform a [planned switchover](#performing-a-planned-switchover) of the write leader node before upgrading that node. We also want to keep the number of planned [switchovers](https://www.enterprisedb.com/docs/pgd/latest/cli/command_ref/pgd_switchover/) to a minimum for availability and efficiency. Accordingly, we will upgrade the current write leader last in our upgrade order, so we only have to perform one planned switchover.
+
+To check to see which node is the current write leader, run the `pgd show-groups` command:
+
+```bash
+sudo -u postgres pgd show-groups
+```
+
+In our example, kaftan, is the current write leader:
+
+```
+Group        Group ID   Type   Parent Group Location Raft Routing Write Leader 
+-----        --------   ----   ------------ -------- ---- ------- ------------ 
+democluster  1935823863 global                       true false                
+dc1_subgroup 1302278103 data   democluster  dc1      true true    kaftan   
+```
+
+The `pgd-show-groups` command shows all groups in the EDB Postgres Distributed cluster and their summary, including the current write leader.
+
+Our upgrade plan can be to upgrade kaboom, upgrade kaolin, perform a planned switchover from kaftan to kaboom, and finally upgrade kaftan.
+
+#### Performing a planned switchover
+
+Since our target node in this walkthrough is kaboom, we won't need to perform a planned switchover yet. 
+
+When you are ready to upgrade kaftan, switchover the write leader from kaftan to kaboom with the following command:
```

Comment:
Confusingly out of order.

Fix it by modifying the output of the show-groups to be kaboom, and then explaining that as kaboom is the write leader we will need to perform a switchover.

Reply:
Hmmm... yeah that is the order I had it in before, but because @MattWalliEDB suggested we would want to upgrade the shadow nodes before the write leader, I opted to have the worked example be a shadow node, as I thought the worked example should be the first node in the upgrade order. However, maybe the worked example can be of the last node to upgrade in the cluster, kaboom, which is the write-leader? I can give that a shot.

Reply:
The flow is all wrong though - there should be advice to target shadow nodes first and a quick way to identify them but you will need to move the write leader on at some point

Reply:
I think I figured out a way with doing the shadow nodes first but also bringing the important content to the general directions.